official/resnet/ctl/ctl_imagenet_main.py

```diff
@@ -127,50 +102,87 @@ def run(flags_obj):
       is_training=True,
       data_dir=flags_obj.data_dir,
       batch_size=flags_obj.batch_size,
-      num_epochs=flags_obj.train_epochs,
       parse_record_fn=parse_record_keras,
       datasets_num_private_threads=flags_obj.datasets_num_private_threads,
       dtype=dtype)
 
+  if strategy:
+    train_ds = strategy.experimental_distribute_dataset(train_ds)
+  
   test_ds = None
   if not flags_obj.skip_eval:
     test_ds = input_fn(
         is_training=False,
         data_dir=flags_obj.data_dir,
         batch_size=flags_obj.batch_size,
-        num_epochs=flags_obj.train_epochs,
         parse_record_fn=parse_record_keras,
         dtype=dtype)
 
     if strategy:
       test_ds = strategy.experimental_distribute_dataset(test_ds)
-      steps_per_eval = IMAGENET_VALIDATION_IMAGES // flags_obj.batch_size
 
-  if strategy:
-    train_ds = strategy.experimental_distribute_dataset(train_ds)
-  steps_per_epoch = APPROX_IMAGENET_TRAINING_IMAGES // flags_obj.batch_size
+  return train_ds, test_ds
+
+
+def get_num_train_iterations(flags_obj):
+  steps_per_epoch = imagenet_main.NUM_IMAGES['train'] // flags_obj.batch_size
   train_epochs = flags_obj.train_epochs
 
   if flags_obj.train_steps:
     train_steps = min(flags_obj.train_steps, steps_per_epoch)
     train_epochs = 1
 
+  steps_per_eval = imagenet_main.NUM_IMAGES['validation'] // flags_obj.batch_size
+  
+  return train_steps, train_epochs, steps_per_eval
+
+def run(flags_obj):
+  """Run ResNet ImageNet training and eval loop using custom training loops.
+
+  Args:
+    flags_obj: An object containing parsed flag values.
+
+  Raises:
+    ValueError: If fp16 is passed as it is not currently supported.
+
+  Returns:
+    Dictionary of training and eval stats.
+  """
+  dtype = flags_core.get_tf_dtype(flags_obj)
+
+  # TODO(anj-s): Set data_format without using Keras.
+  data_format = flags_obj.data_format
+  if data_format is None:
+    data_format = ('channels_first'
+                   if tf.test.is_built_with_cuda() else 'channels_last')
+  tf.keras.backend.set_image_data_format(data_format)
+
+  strategy = distribution_utils.get_distribution_strategy(
+      distribution_strategy=flags_obj.distribution_strategy,
+      num_gpus=flags_obj.num_gpus,
+      num_workers=distribution_utils.configure_cluster(),
+      all_reduce_alg=flags_obj.all_reduce_alg,
+      num_packs=flags_obj.num_packs)
+
+  train_ds, test_ds = get_input_dataset(flags_obj, strategy)
+  train_steps, train_epochs, steps_per_eval = get_num_train_iterations(flags_obj)
+
+  time_callback = keras_utils.TimeHistory(flags_obj.batch_size, 1)
```

Comment:
The second argument should take the flag value? FLAGS.log_steps

Reply:
Done.